industrial_ci/src/tests/abi_check.sh

```diff
@@ -0,0 +1,145 @@
+    catkin build -w "$workspace"
+
+    mkdir "$workspace/abi_dumps"
+    for l in "$workspace/install/lib"/*.so; do
+        abi-dumper "$l" -ld-library-path "/opt/ros/$ROS_DISTRO/lib" -o "$workspace/abi_dumps/$(basename "$l" .so).dump" -lver "$version" -public-headers "$workspace/install/include"
+    done
+}
+
+function abi_setup_rosdep() {
+    # Setup rosdep
+    rosdep --version
+    if ! [ -d /etc/ros/rosdep/sources.list.d ]; then
+        sudo rosdep init
+    fi
+    ret_rosdep=1
+    rosdep update || while [ $ret_rosdep != 0 ]; do sleep 1; rosdep update && ret_rosdep=0 || echo "rosdep update failed"; done
+    rosdep install -q --from-paths "$TARGET_REPO_PATH" --ignore-src -y > /dev/null
+}
+
+function run_abi_check() {
+    export ABICHECK_VERSION
+    export ABICHECK_URL
+
+    if [ -z "$ABICHECK_URL" ]; then
+        local ref_list
+        if ref_list=($(cd "$TARGET_REPO_PATH" && git rev-list --parents -n 1 HEAD)) && [ "${#ref_list[@]}" -gt 2 ]; then
+            ABICHECK_URL="#${ref_list[1]}"
+            ABICHECK_VERSION="${ref_list[1]}"
+        elif [ -z "$ABICHECK_FALLBACK" ]; then
+            error "Could not find commit parents, please set ABICHECK_FALLBACK"
+        else
+            ABICHECK_URL="$ABICHECK_FALLBACK"
+        fi
+    fi
+
+    if [ -z "$ABICHECK_VERSION" ]; then
+        local target_ext
+        target_ext=$(grep -Pio '\.(zip|tar\.\w+|tgz|tbz2)\Z' <<< "${ABICHECK_URL%%\?*}")
+        ABICHECK_VERSION=$(basename "$ABICHECK_URL" "$target_ext")
+    fi
+
+    ici_require_run_in_docker # this script must be run in docker
+
+    ici_time_start install_abi_compliance_checker
+    abi_install > /dev/null
+    ici_time_end  # install_abi_compliance_checker
+
+    ici_time_start setup_rosdep
+    abi_setup_rosdep
+    ici_time_end  # setup_rosdep
+
+    ici_time_start abi_build_new
+    mkdir -p "/abicheck/new/src"
+    ln -s "$TARGET_REPO_PATH" "/abicheck/new/src"
+    abi_build_workspace /abicheck new
+    ici_time_end  # abi_build_new
+
+    ici_time_start abi_build_old
+    abi_prepare_src "/abicheck/old/$ABICHECK_VERSION/src" "$ABICHECK_URL"
+    abi_build_workspace /abicheck/old "$ABICHECK_VERSION"
+    ici_time_end  # abi_build_old
+
+    local reports_dir="/abicheck/reports/$ABICHECK_VERSION"
+    mkdir -p "$reports_dir"
+
+    local broken=()
+    for n in /abicheck/new/abi_dumps/*.dump; do
+        local l=$(basename "$n" ".dump")
+        local o="/abicheck/old/$ABICHECK_VERSION/abi_dumps/$l.dump"
+        if [ -f "$o" ]; then
+            ici_time_start "abi_check_$l"
+            local ret=0
+            abi-compliance-checker -report-path "$reports_dir/$l.html" -l "$l" -n "$n" -o "$o" || ret=$?
+            if [ "$ret" -eq "0" ]; then
+                ici_time_end # abi_check_*
+            elif [ "$ret" -eq "1" ]; then
+                links -dump "$reports_dir/$l.html"
```

Comment:
So in the case of [a failing CI job](https://travis-ci.org/ros-industrial/industrial_ci/jobs/282457731) (which is a very nice addition!),
https://travis-ci.org/ros-industrial/industrial_ci/jobs/282457731#L1548
```
Preparing, please wait ...
Comparing ABIs ...
Comparing APIs ...
Creating compatibility report ...
Binary compatibility: 100%
Source compatibility: 100%
Total binary compatibility problems: 0, warnings: 0
Total source compatibility problems: 0, warnings: 0
Report: /abicheck/reports/0.7.1/libtopic_to_socketcan.html
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Function abi_check_libtopic_to_socketcan returned with code '0' after 0 min 0 sec

Broken libraries: libcanopen_motor libcanopen_ros_chain
:
```

It'd be really great if we can see in the CI output for why the ABI check failed. 
Assuming some info is available in this report html file, we might want to `cat` it?

Reply:
The last section is just a summary.
The details can be found in the previous folds, the broken ones are yellow instead of green :)